app/components/summary_card/view.html.erb

```diff
@@ -3,23 +3,21 @@
     <h<%= heading_level %> class="app-summary-card__title">
       <%= title %>
     </h<%= heading_level %>>
-    
+
     <% if header_actions %>
       <div class="app-summary-card__actions">
         <%= header_actions %>
-      </div>      
+      </div>
     <% end %>
   </header>
   <div class="app-summary-card__body">
     <%= render GovukComponent::SummaryList.new do |component| %>
       <% rows.each do |row| %>
-        <% component.slot(
-             :row,
-             key: row[:key],
-             value: row[:value],
-             action: row[:action],
-             classes: row[:key].to_s.downcase.gsub(/ /, "-"),
-             ) %>
+        <%
+          slot_args = { key: row[:key], value: row[:value], classes: row[:key].to_s.downcase.gsub(/ /, "-") }
+          slot_args.merge!(action: row[:action]) unless prevent_action?
```

Comment:
Should this logic be in the component as a method that receives the args and returns the value for the view? I'm generally weary of doing any sort of variable assignment in views to keep them as dumb as possible

Reply:
You are right! I should have known better.